flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/MatchRecognizeITCase.scala

```diff
@@ -416,6 +416,54 @@ class MatchRecognizeITCase(backend: StateBackendMode) extends StreamingWithState
     assertEquals(expected.sorted, sink.getAppendResults.sorted)
   }
 
+  @Test
+  def testPartitionByWithParallelSource(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    env.setParallelism(2)
+    val tEnv = StreamTableEnvironment.create(env, TableTestUtil.STREAM_SETTING)
+
+    val data = new mutable.MutableList[(String, Long, Int, Int)]
+    data.+=(("ACME", 1L, 19, 1))
+    data.+=(("ACME", 2L, 17, 2))
+    data.+=(("ACME", 3L, 13, 3))
+    data.+=(("ACME", 4L, 20, 4))
+    val parallelSource = new ParallelCollectionWrapper[(String, Long, Int, Int)](
+      data, 0, data.length)
+
+    val t = env.fromParallelCollection(parallelSource)
+      .assignAscendingTimestamps(tickerEvent => tickerEvent._2)
```

Comment:
Apply `.setParallelism(4)` to reshuffle data to reproduce the problem, in this way, we don't need to introduce `ParallelCollectionWrapper`.